lib/livebook/update_check.ex

```diff
@@ -0,0 +1,146 @@
+defmodule Livebook.UpdateCheck do
+  @moduledoc false
+
+  # Periodically checks for available Livebook update.
+
+  use GenServer
+
+  require Logger
+
+  @name __MODULE__
+  @timeout :infinity
+
+  @hour_in_ms 60 * 60 * 1000
+  @day_in_ms 24 * @hour_in_ms
+
+  @doc false
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, {}, name: @name)
+  end
+
+  @doc """
+  Returns the latest Livebook version if it's more recent than the
+  current one.
+  """
+  @spec new_version() :: String.t() | nil
+  def new_version() do
+    GenServer.call(@name, :get_new_version, @timeout)
+  end
+
+  @doc """
+  Returns whether the update check is enabled.
+  """
+  @spec enabled?() :: boolean()
+  def enabled?() do
+    GenServer.call(@name, :get_enabled, @timeout)
+  end
```

Comment:
@josevalim since we want the settings page to call `Updatecheck.set_enabled`, I added the read for consistency. We need to keep track of whether the check is enabled inside the process, so this is pretty much caching of whatever we have in storage.

On a related note, I don't think we get much by using `:persistent_term` here, so I made it into a call too. Let me know if you have any concerns :)